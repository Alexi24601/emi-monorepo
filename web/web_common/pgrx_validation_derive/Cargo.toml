[package]
name = "pgrx_validation_derive"
version = "0.1.0"
edition = "2021"

[dependencies]
syn.workspace = true
quote.workspace = true
proc-macro2.workspace = true

[dev-dependencies]
trybuild.workspace = true
validation-errors.workspace = true

[lib]
proc-macro = true

[features]
pgrx = []
