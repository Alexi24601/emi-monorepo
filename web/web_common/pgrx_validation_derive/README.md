# PGRX Validation Derive

Procedural macros to dispatch the function output data type according to the features enabled.
