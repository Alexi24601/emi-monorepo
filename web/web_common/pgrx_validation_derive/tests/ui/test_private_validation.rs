//! Submodule for testing compile failure of the `validation` attribute on private functions.
use pgrx_validation_derive::validation;

#[validation]
/// A simple private validation function which should fail.
fn is_not_empty(string: &str) -> Result<(), String> {
    if string.is_empty() {
        Err("String is empty".to_string())
    } else {
        Ok(())
    }
}

/// The main for this test.
fn main() {}
