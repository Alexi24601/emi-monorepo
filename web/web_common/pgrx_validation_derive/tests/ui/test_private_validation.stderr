error: Function `is_not_empty` must be public to be decorated with `validation` - add a `pub` keyword before the function definition if appropriate.
 --> tests/ui/test_private_validation.rs:4:1
  |
4 | #[validation]
  | ^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `validation` (in Nightly builds, run with -Z macro-backtrace for more info)
